Jenkinsfile

```diff
@@ -402,14 +402,13 @@ stage('Pre-build sources') {
 		}
 	}
 }
-
-stage('Default build') {
-	if (!enableDefaultBuild) {
-		echo 'Skipping default build and integration tests in the default environment'
-		helper.markStageSkipped()
-		return
-	}
-	runBuildOnNode( NODE_PATTERN_BASE, [time: 2, unit: 'HOURS'] ) {
+runBuildOnNode( NODE_PATTERN_BASE, [time: 2, unit: 'HOURS'] ) {
```

Comment:
Each `runBuildOnNode` instruction will potentially pick another node with the same label. You need to put both `stage` blocks inside the same `runBuildOnNode` block.